apps/src/templates/progress/ProgressBubble.jsx

```diff
@@ -148,7 +155,12 @@ class ProgressBubble extends React.Component {
       const paramString = queryString.stringify(queryParams);
       href = url;
       if (paramString.length > 0) {
-        href += '?' + paramString;
+        // If href already has 1 or more query params, our delimiter will be '&'.
+        // If href has no query params, our delimiter is '?'.
+        // TODO: (madelynkasula) Refactor this logic to use queryString.parseUrl(href)
+        // instead. Our current version of query-string (4.1.0) does not yet have this method.
+        const delimiter = /\?/.test(href) ? '&' : '?';
+        href += delimiter + paramString;
```

Comment:
this looks like it will work just fine. however, if you want to stay out of the business of query string parsing, you could parse the url using the `URL` class:
```js
const url = new URL(href);
queryParams.merge(queryString.parse(url.search);
url.search = queryString.stringify(queryParams);
href = url.href;
```

Reply:
haha yes, i do! TIL about the URL class 😄i'll refactor to use that instead

Reply:
very strange, but when i updated this code to use the URL class, all of the unit tests failed because `new URL(href)` always evaluates to an empty object. this is only happening in unit tests, not in the browser. i don't want to spend too much time on it, so i'm going to leave the code as-is for now

Reply:
i want to come back to this later, so adding an example:

in the browser:
```
new URL('http://some.url/foo/bar')
=> URL {href: "http://some.url/foo/bar", origin: "http://some.url", protocol: "http:", username: "", password: "", …}
```

from a unit test file (i.e., [`ProgressBubbleTest.js`](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/test/unit/templates/progress/ProgressBubbleTest.js)):
```
new URL('http://some.url/foo/bar')
=> {}
```

Reply:
That seems fine for now. Thanks for trying! Kudos for including comments on how to alter this once `query-string` is upgraded.